.circleci/config.yml

```diff
@@ -28,7 +28,7 @@ jobs:
             echo ${CI_PULL_REQUEST//*pull\//} | tee merge.txt
             if [[ $(cat merge.txt) != "" ]]; then
               echo "Merging $(cat merge.txt)";
-              git remote add upstream https://github.com/scipy/scipy.git;
+              git remote add upstream git://github.com/scipy/scipy.git;
```

Comment:
spurious change?

May be related to merging master into this branch. Unless you want this PR to be squash-merged, it'd be good to rebase rather than merge master.

Reply:
okay, this is re-adding a whole bunch of stuff, can you get rid of the commit and rebase so that the diff only shows sparse changes?

Reply:
As @jarrodmillman pointed out, this crossed wires with the PR to make private API use underscores. May need a bit of git fu (or manual redoing of diff) to update in case just removing the merge with master doesn't work.

Reply:
Yeah, the problem is that Gagandeep's PR to make things private touched basically single file in scipy.sparse, making it a complete mess to do a rebase (conflicts all along the way).  I therefore took the strategy of cherry-picking his commits and then merging scipy into mine, using the "prefer mine" strategy.  I'm open to tips on how to handle this better.

Reply:
Is there a reason not to squash your commits? There's 9 commits here, and only the first couple are large-ish, the rest are just fix up commits. If you squash them first, you can do a single rebase. 

Alternatively, now that you've done the cherry-picking and merging, just `git reset --mixed <commit-id>` to get a clean diff on top of master, then recommit that.

Reply:
For now, went with squashing all my changes into a single commit and rebasing.